src/server.ts

```diff
@@ -749,3 +753,77 @@ function findCreatedAccountSequenceInTransactionMeta(
 
   throw new Error("No account created in transaction");
 }
+
+
+// TODO - remove once rpc updated to
+// append expiration entry per data LK requested onto server-side response 
+// https://github.com/stellar/soroban-tools/issues/1010
+async function mergeResponseExpirationLedgers(response: Promise<SorobanRpc.RawGetLedgerEntriesResponse>,
+  requestedKeys: xdr.LedgerKey[]
+): Promise<SorobanRpc.RawGetLedgerEntriesResponse> {
+  let ledgerEntriesResponse = await response;
+  const expirationKeyToRawEntryResult:Map<String, SorobanRpc.RawLedgerEntryResult> = new Map();
+  const requestedKeyXdrs = new Set<String>(requestedKeys.map(requestedKey =>
+    requestedKey.toXDR('base64')));
+
+  (ledgerEntriesResponse.entries ?? []).forEach(rawEntryResult => {
+    if (!rawEntryResult.key) {
+      // don't interpret here, just pass it through to the outgoing response as-is
+      expirationKeyToRawEntryResult.set(Math.random().toString(), rawEntryResult)
+      return;
+    }
+    const parsedKey:xdr.LedgerKey = xdr.LedgerKey.fromXDR(rawEntryResult.key, 'base64');
+    
+    if (parsedKey.switch().value !== xdr.LedgerEntryType.expiration().value ||
+        requestedKeyXdrs.has(rawEntryResult.key)) {
+      let keyHash = hash(parsedKey.toXDR()).toString()
+      let rawEntryResultLookup = expirationKeyToRawEntryResult.get(keyHash);
+
+      if (!rawEntryResultLookup) {
+        rawEntryResultLookup = rawEntryResult;
+        expirationKeyToRawEntryResult.set(keyHash, rawEntryResultLookup);
+      } 
+      rawEntryResultLookup.key = rawEntryResult.key;
+      rawEntryResultLookup.xdr = rawEntryResult.xdr;
+      return;
+    } 
+
+    let keyHash = parsedKey.expiration().keyHash().toString();
+    let rawEntryResultLookup = expirationKeyToRawEntryResult.get(keyHash);
+
+    if (!rawEntryResultLookup) {
+      rawEntryResultLookup = {
+        key: "", 
+        xdr: ""
+      };
+      expirationKeyToRawEntryResult.set(keyHash, rawEntryResultLookup);
+    }
+    rawEntryResultLookup.expiration = xdr.LedgerEntryData
+      .fromXDR(rawEntryResult.xdr, 'base64')
+      .expiration()
+      .toXDR('base64');
+  });
+  
+  ledgerEntriesResponse.entries = [...expirationKeyToRawEntryResult.values()];
+  return ledgerEntriesResponse;
+}
+
+// TODO - remove once rpc updated to
+// include expiration entry on responses for any data LK's requested  
+// https://github.com/stellar/soroban-tools/issues/1010
+function expandRequestIncludeExpirationLedgers(
+  keys: Array<xdr.LedgerKey>
+): Array<xdr.LedgerKey> {
+  let includingExpiryKeys: Array<xdr.LedgerKey> = new Array()
+  keys.forEach(key => {
+    if (key.switch().value !== xdr.LedgerEntryType.expiration().value) {
+      const expirationKey = xdr.LedgerKey.expiration(
+        new xdr.LedgerKeyExpiration(
+          { keyHash: hash(key.toXDR())}
+      ));
+      includingExpiryKeys.push(expirationKey);
+    }
+    includingExpiryKeys.push(key);
+  });
+  return includingExpiryKeys;
```

Comment:
Since we are always returning the entire `keys` array, it might be a little more idiomatic to do concat + filter + map like so:

```typescript
return keys.concat(keys
  .filter(key => { /* exclude expiration keys */ })
  .map(key => { /* do hashing stuff */ });
```